### PR TITLE
[ISSUE#4917] Optimize the retry logic in appending a no-op request when controller change to leader

### PR DESCRIPTION
**Pull request by TheR1sing3un**
…t to dledger when the controller change to leader

1. optimize the retry logic in appending a empty request to dledger when
the controller change to leader

link issue: https://github.com/apache/rocketmq/issues/4917

**Make sure set the target branch to `develop`**

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java [hunk 1]
```diff
@@ -432,6 +432,16 @@ public void handle(long term, MemberState.Role role) {
...
+                                    break;
+                                }
+                                log.error("Controller leader append initial log failed, try again");
+                                tryTimes++;
+                                if (tryTimes % 3 == 0) {
+                                    log.warn("Controller leader append initial log failed too many times, please wait a while");
+                                }
                             } catch (final Throwable e) {
                                 log.error("Error happen when controller leader append initial request to dledger", e);
                                 tryTimes++;
```
Comment (RongtongJin) [inline controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java:447]: If an exception is thrown, do we need to determine whether it is a leader?

Comment (TheR1sing3un) [reply controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java:447]: Yep~ Maybe we can check the state once the appending is failed(appendToDLedfer return false or appendToDLedger AndWait throws an exception)

Comment (RongtongJin) [reply controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java]: > Yep~ Maybe we can check the state once the appending is failed(appendToDLedfer return false or appendToDLedger AndWait throws an exception)

How about creating a function to deal with it？

Comment (TheR1sing3un) [reply controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java]: > > Yep~ Maybe we can check the state once the appending is failed(appendToDLedfer return false or appendToDLedger AndWait throws an exception)
> 
> How about creating a function to deal with it？

got it!

**Review by hzh0425 — APPROVED** [review]

File: controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java (hunk 1 above)
Comment (TheR1sing3un) [reply controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java]: > > Yep~ Maybe we can check the state once the appending is failed(appendToDLedfer return false or appendToDLedger AndWait throws an exception)
> 
> How about creating a function to deal with it？

I find that maybe it is uncessary to add a function to deal with it.

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**